index.bs

```diff
@@ -1409,6 +1958,44 @@ partial interface MLGraphBuilder {
     computed as the sum of all the input sizes of the same dimension.
 </div>
 
+<details open>
+  <summary>
+    The {{MLGraphBuilder/concat(inputs, axis)}} steps are:
+  </summary>
+  <div algorithm=concat class=algorithm-steps>
+    <div class="note">
+        The permissions and context validity have been checked by [[#api-mlgraphbuilder-constructor]] steps.
+    </div>
+    1. [=Assert=]: the type of |inputs| is sequence of {{MLOperand}} objects.
+    1. [=Assert=]: the type of |axis| is `unsigned long`.
+    1. [=Assert=]: the shape, i.e. {{MLOperandDescriptor/dimensions}}) of each operand in |inputs| is the same, except on the dimension given by |axis| on which they are concatenated.
+    1. [=Assert=]: the {{MLOperandDescriptor/type}} of each operand in |inputs| is the same.
+    1. If any of the following steps fail, then [=exception/throw=] a "{{DataError}}" {{DOMException}}.
+        1. If |axis| is greater than or equal to the <a>rank</a> of |inputs|, fail.
+        1. Let |desc| be |inputs|[0].{{MLOperand/[[descriptor]]}}.
+        1. Let |desc|.{{MLOperandDescriptor/dimensions}}[|axis|] be `0`.
+        1. [=map/For each=] |index| between 0 and the <a>rank</a> of |inputs|:
+            1. If <a>validating MLOperand</a> given |inputs|[|index|] and [=this=] returns `false`, then fail.
+            1. [=map/For each=] |dim| between 0 and the <a>rank</a> of |inputs|[|index|]:
+                <div class="note">
+                    If the shape of each corresponding dimension and type of the operands, except for those of the dimension given by |axis|, is not the same, fail.
+                </div>
+                1. If |dim| is not equal to |axis| and if |inputs|[|index|].{{MLOperandDescriptor/dimensions}}[|dim|] is not equal to |inputs|[0].{{MLOperandDescriptor/dimensions}}[|dim|], fail.
+                1. If |inputs|[|dim|].{{MLOperandDescriptor/type}} is not equal to |inputs|[0].{{MLOperandDescriptor/type}}.
```

Comment:
`|inputs|[|index|].{{MLOperandDescriptor/type}}`? and move it upper above line 1979? I guess it's to validate each input operand of Concat.

Reply:
Could you help explain why is `|inputs|[|dim|].{{MLOperandDescriptor/type}}` ? Thanks!

Reply:
Sorry, looks like I forgot to fix this one. Moved the line in question up, and fixed the index.